src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,211 @@
+import { PersonRow } from '../PersonRow';
+import { Person } from '../../types';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { Dispatch, SetStateAction } from 'react';
+import { ErrorType } from '../../types/ErrorType';
+import { SearchParams, getSearchWith } from '../../utils/searchHelper';
+import cn from 'classnames';
+import { SortField } from '../../pages/SortTypes';
+
+interface Params {
+  people: Person[];
+  setError: Dispatch<SetStateAction<ErrorType>>;
+}
+
+export const PeopleTable: React.FC<Params> = ({ people, setError }) => {
+  const { slug } = useParams();
+  const normalizedSlug = slug || null;
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || null;
+  const query = searchParams.get('query') || null;
+  const centuries = searchParams.get('centuries') || null;
+  const sortField = searchParams.get('sort');
+  const sortOrder = searchParams.get('order');
+
+  const setSearchWith = (params: SearchParams) => {
+    setSearchParams(getSearchWith(searchParams, params));
+  };
+
+  // FILTERING
+  const visiblePeople = [...people].filter(person => {
+    // check if sex was set and filter people by sex
+    if (sex && person.sex !== sex) {
+      return false;
+    }
+
+    // checks if query was set and filters people by their name, father name and mother name
+    if (
+      query &&
+      !person.name.includes(query) &&
+      !person.motherName?.includes(query) &&
+      !person.fatherName?.includes(query)
+    ) {
+      return false;
+    }
+
+    // checks if born or death year is within certain century
+    if (
+      centuries &&
+      !centuries.includes((person.born % 100).toString()) &&
```

Comment:
I believe there should be something like this: `centuries.includes(${Math.ceil(person.born` / `100)})`